File: cpp/lv1/inner_product.cpp
/* Answer 1.

#include <string>
#include <vector>

using namespace std;

int solution(vector<int> a, vector<int> b) {
    int answer = 0;
    
    for(int i = 0; i < a.size(); i++)
    {
        answer += a[i] * b[i];
    }
    
    return answer;
} */


/* Answer 2.
내적 함수 inner_product 사용
두 개의 시작 위치를 인자로 받아 각 벡터의 원소를 곱하고 더한 결과를 반환.

#include <vector>
#include <numeric>
  
using namespace std;

int solution(vector<int> a, vector<int> b) {    
    return inner_product(a.begin(),a.end(),b.begin(),0);
} */

File: python/lv1/inner_product.py
# 내적

'''
Answer 1.

def solution(a, b):
    answer = 0
    
    for i in range(len(a)):
        answer += a[i] * b[i]
    
    return answer
'''

'''
Answer 2.

def solution(a, b):
    answer = 0
    
    for x, y in zip(a,b):
        answer += x * y
        
    return answer

'''

'''
Answer 3.

def solution(a, b):

    return sum([x*y for x, y in zip(a,b)])
'''
